python/lsst/ts/adamSensors/adamSensorsCSC.py

```diff
@@ -26,23 +34,98 @@ def __init__(self, config_dir=None, initial_state=salobj.State.STANDBY, initial_
     async def begin_start(self, id_data):
         await super().begin_start(id_data)
         self.adam = AdamModel(self.log, simulation_mode=True)
-        await self.adam.connect(self.config.ip, self.config.port)
+        loop = asyncio.get_event_loop()
+        executor = concurrent.futures.ThreadPoolExecutor()
+        await loop.run_in_executor(executor, self.adam.connect, self.config.adam_ip, self.config.adam_port)
+        self.connected = True
+        await self.telemetry_loop()
```

Comment:
You are awaiting for `telemetry_loop` to finish execution here. This will block until `telemetry_loop` finishes executing. But there is a `while self.connected` in that coroutine so it will block here while `self.connected==True`, which is probably not what you want. You probably want to create a task in the background and assign it to some attribute so you can cancel it later, e.g.;

```
self.telemetry_loop_task = asyncio.create_task(self.telemetry_loop())
```